SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverGameplayDebuggerContext.cpp

```diff
@@ -242,6 +255,84 @@ void USpatialNetDriverGameplayDebuggerContext::TickServer()
 
 		It.RemoveCurrent();
 	}
+
+	for (auto& Pair : TrackedEntities)
+	{
+		if (Pair.Value.Component.TrackPlayer)
+		{
+			if (!NetDriver->HasServerAuthority(Pair.Key))
+			{
+				// Do not track player if we don't have authority
+				continue;
+			}
+
+			TWeakObjectPtr<AGameplayDebuggerCategoryReplicator> ReplicatorWeakObjectPtr = Pair.Value.ReplicatorWeakObjectPtr;
+			AGameplayDebuggerCategoryReplicator* Replicator = ReplicatorWeakObjectPtr.Get();
+			if (Replicator == nullptr)
+			{
+				// Not unexpected - assume latency and wait for actor to appear
+				continue;
+			}
+
+			const AActor* PlayerControllerActor = Cast<AActor>(Replicator->GetReplicationOwner());
+			if (!ensureMsgf(PlayerControllerActor != nullptr, TEXT("Serious error: Cannot find player controller for replicated gameplay debugger actor")))
+			{
+				continue;
+			}
+
+			check(NetDriver->LoadBalanceStrategy);
+			auto GetActorVirtualWorkerId = [](const AActor& InActor, const USpatialNetDriver& InNetDriver, const SpatialGDK::FSubView& InSubView)->VirtualWorkerId
+			{
+				check(InNetDriver.PackageMap);
+
+				const Worker_EntityId EntityId = InNetDriver.PackageMap->GetEntityIdFromObject(&InActor);
+				if (!ensureMsgf(EntityId != SpatialConstants::INVALID_ENTITY_ID, TEXT("Serious error: Cannot find actor in package map")))
+				{
+					return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+				}
+
+				for (const SpatialGDK::ComponentData& Data : InSubView.GetView()[EntityId].Components)
+				{
+					if (Data.GetComponentId() == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
+					{
+						SpatialGDK::AuthorityIntent Intent = SpatialGDK::AuthorityIntent(Data.GetUnderlying());
+						const VirtualWorkerId VirtualWorkerId = Intent.VirtualWorkerId;
+						return VirtualWorkerId;
+					}
+				}
+
+				return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+			};
+
+			check(NetDriver && SubView);
+			const VirtualWorkerId PlayerControllerVirtualWorkerId = GetActorVirtualWorkerId(*PlayerControllerActor, *NetDriver, *SubView);
```

Comment:
I think this would benefit from a comment that explains what is going on here: Since we are tracking the player, we look up the virtual worker ID that it is currently assigned to by the load balancer. We then try to make sure that our own authority intent for the replicator actor itself matches that, or otherwise update it if necessary.